deepmd/train/trainer.py

```diff
@@ -829,22 +913,27 @@ def train(self, train_data=None, valid_data=None):
                         train_batches = {}
                         valid_batches = {}
                         # valid_numb_batch_dict
-                        for fitting_key in train_data:
-                            train_batches[fitting_key] = [
-                                train_data[fitting_key].get_batch()
+                        for fitting_key_ii in train_data:
```

Comment:
Is it really necessary to change the name of iterator?

Reply:
Yes, it is necessary. Otherwise, it will cover the `fitting_key` for `valid_on_the_fly`.